include/seqan/modifier/modifier_padding.h

```diff
@@ -0,0 +1,413 @@
+inline auto
+operator*(ModifiedIterator<THost, ModPadding> & me)
+{
+    return (SEQAN_UNLIKELY(atEnd(host(me)))) ? cargo(me)._paddedValue : *host(me);
+}
+
+template <typename THost>
+inline auto
+operator*(ModifiedIterator<THost, ModPadding> const & me)
+{
+    return (SEQAN_UNLIKELY(atEnd(host(me)))) ? (cargo(me)._paddedValue) : (*host(me));
+}
+
+// ----------------------------------------------------------------------------
+// Function operator++()
+// ----------------------------------------------------------------------------
+
+template <typename THost>
+inline ModifiedIterator<THost, ModPadding> &
+operator++(ModifiedIterator<THost, ModPadding> & me)
+{
+    if (SEQAN_UNLIKELY(atEnd(host(me))))
+        --cargo(me)._remainingSteps;
+    else
+        ++host(me);
+    return me;
+}
+
+// ----------------------------------------------------------------------------
+// Function operator+=()
+// ----------------------------------------------------------------------------
+
+template <typename THost, typename TSize>
+inline ModifiedIterator<THost, ModPadding> &
+operator+=(ModifiedIterator<THost, ModPadding> & me, TSize const steps)
+{
+    if (SEQAN_UNLIKELY(atEnd(host(me))))
+    {
+        cargo(me)._remainingSteps -= steps;   // Remove 'steps' from remaining steps.
+    }
+    else
+    {
+        auto rem = (end(container(host(me)), Rooted()) - host(me));
+        if (SEQAN_LIKELY(static_cast<decltype(rem)>(steps) <= rem))  // Move host by 'steps' forward.
+        {
+            host(me) += steps;
+        }
+        else  // Move host by 'rem' forward and remove diff from cargo.
+        {
+            host(me) += rem;
+            cargo(me)._remainingSteps -= (steps - rem);
+        }
+    }
+    return me;
+}
+
+// ----------------------------------------------------------------------------
+// Function operator--()
+// ----------------------------------------------------------------------------
+
+template <typename THost>
+inline ModifiedIterator<THost, ModPadding> &
+operator--(ModifiedIterator<THost, ModPadding> & me)
+{
+    if (SEQAN_LIKELY(cargo(me)._remainingSteps == cargo(me)._expandedSize))
+        --host(me);
+    else
+        ++cargo(me)._remainingSteps;
+    return me;
+}
+
+// ----------------------------------------------------------------------------
+// Function operator-=()
+// ----------------------------------------------------------------------------
+
+template <typename THost, typename TSize>
+inline ModifiedIterator<THost, ModPadding> &
+operator-=(ModifiedIterator<THost, ModPadding> & me, TSize const steps)
+{
+    if (SEQAN_UNLIKELY(atEnd(host(me))))
+    {
+        auto rem = cargo(me)._expandedSize - cargo(me)._remainingSteps;
+        if (static_cast<decltype(rem)>(steps) <= rem)
+        {
+            cargo(me)._remainingSteps += steps;
+        }
+        else
+        {
+            cargo(me)._remainingSteps = cargo(me)._expandedSize;
+            host(me) -= steps - rem;
+        }
+    }
```

Comment:
this paragraph looks trange to me, IMO _remainingSteps can never be _expandedSize unless the host container is empty.. but maybe I misunderstood something?